sebsync.py

```diff
@@ -126,54 +131,89 @@ def get_remote_ebooks() -> None:
             id=entry.find("dc:identifier", ns).text,
             title=entry.find("atom:title", ns).text,
             author=entry.find("atom:author", ns).find("atom:name", ns).text,
-            href=entry.find(f".//atom:link[@title='{type_selector[options.type]}']", ns).attrib[
-                "href"
-            ],
+            href=entry.find(
+                f".//atom:link[@title='{type_selector[options.type]}']", ns
+            ).attrib["href"],
             updated=fromisoformat(entry.find("atom:updated", ns).text),
         )
         remote_ebooks[remote_ebook.id] = remote_ebook
     if not remote_ebooks:
-        raise click.ClickException("OPDS catalog download failed. Is email address correct?")
+        raise click.ClickException(
+            "OPDS catalog download failed. Is email address correct?"
+        )
     if options.verbose:
         click.echo(f"Found {len(remote_ebooks)} remote ebooks.")
 
 
-def get_local_ebooks() -> None:
+def get_local_ebooks(local_cache: dict) -> None:
     """Retrieve metadata of Standard EPUBs in the specified directory and subdirectories."""
-    for path in options.books.glob("**/*.epub"):
-        if not path.is_file():
-            continue
-        try:
-            with zipfile.ZipFile(path) as zip:
-                with zip.open("META-INF/container.xml") as file:
-                    root = ElementTree.parse(file)
-                    ns = {"container": "urn:oasis:names:tc:opendocument:xmlns:container"}
-                    rootfile = root.find(".//container:rootfile", ns).attrib["full-path"]
-                with zip.open(rootfile) as file:
-                    root = ElementTree.parse(file)
-                    ns = {
-                        "opf": "http://www.idpf.org/2007/opf",
-                        "dc": "http://purl.org/dc/elements/1.1/",
-                    }
-                    metadata = root.find("opf:metadata", ns)
-                    id = metadata.find("dc:identifier", ns)
-                    if id is None or "standardebooks.org" not in id.text:
-                        continue
-                    modified = metadata.find(".//opf:meta[@property='dcterms:modified']", ns)
+    if options.type == "kindle":
+
+        # Retrieve the id, title, modified date, and ETag from our cache file
+        for path in options.books.glob("**/*.azw3"):
+            if not path.is_file():
+                continue
+            try:
+                filename = os.path.basename(path)
+                if filename in local_cache:
```

Comment:
It's highly likely that a book filename will be moved or renamed after being downloaded. And when that happens, under this scheme, it seems Kindle books would become untraceable to sebsync. This is why I was previously mentioning the use of file hashes. If a Kindle book is downloaded by sebsync, at that time, it can store the ([SHA-256?](https://docs.python.org/3/library/hashlib.html#file-hashing)) hash and the Standard Ebooks identifier together, so that if the file is renamed, it can still be located in future runs.

Reply:
Ahh, I see. When you were previously mentioning file hashes I was thinking of them to help identify changes between the local and remote files, hence my thought on saving ETags upon download and using them to flag when the remote file differed. I'll see what I can do for using file hashes to keep track of local files.